src/sidebar/components/PendingUpdatesButton.tsx

```diff
@@ -0,0 +1,54 @@
+import { IconButton, RefreshIcon } from '@hypothesis/frontend-shared/lib/next';
+import { useEffect } from 'preact/hooks';
+
+import { withServices } from '../service-context';
+import type { ToastMessengerService } from '../services/toast-messenger';
+import { useSidebarStore } from '../store';
+
+export type PendingUpdatesButtonProps = {
+  onClick: () => void;
+
+  // Injected
+  toastMessenger: ToastMessengerService;
+};
+
+function PendingUpdatesButton({
+  onClick,
+  toastMessenger,
+}: PendingUpdatesButtonProps) {
+  const store = useSidebarStore();
+  const pendingUpdateCount = store.pendingUpdateCount();
+  const hasPendingUpdates = store.hasPendingUpdates();
+
+  useEffect(() => {
+    if (hasPendingUpdates) {
+      toastMessenger.success(
+        `There are ${pendingUpdateCount} new annotations.`,
```

Comment:
Because we will only announce once and not again until any additional annotations are loaded in, I think we might consider not announcing the specific count, as it may not be accurate by the time the user loads in the annotations. 

```suggestion
        `New annotations are available`,
```

This will read better when it can be followed with "Press (whatever) to load these annotations". And even better down the line when we can confirm with "X new annotations loaded".

Reply:
BTW, I see now why you were a little ambivalent about adding a `hasPendingUpdates` selector, since this component also needs `pendingUpdateCount`. But I still think it's fine...

Reply:
> BTW, I see now why you were a little ambivalent about adding a hasPendingUpdates selector, since this component also needs pendingUpdateCount. But I still think it's fine...

Yes. I mean, we could have a locally memoized flag which depends on the amount. Something in these lines:

```tsx
const hasPendingUpdates = useMemo(() => pendingUpdateCount > 0, [pendingUpdateCount]);
```

But if we are going to do that we may as well have the selector, which is easier to reason about and can be used somewhere else if needed.